src/profile-logic/marker-data.js

```diff
@@ -119,71 +119,184 @@ export function getSearchFilteredMarkerIndexes(
   getMarker: (MarkerIndex) => Marker,
   markerIndexes: MarkerIndex[],
   markerSchemaByName: MarkerSchemaByName,
-  searchRegExp: RegExp | null,
+  searchRegExps: MarkerRegExps | null,
   categoryList: CategoryList
 ): MarkerIndex[] {
-  if (!searchRegExp) {
+  if (!searchRegExps) {
     return markerIndexes;
   }
+
+  const hasPositiveRegexps =
+    searchRegExps.generic ||
+    Array.from(searchRegExps.fieldMap.entries()).some(
+      ([_, val]) => val.positive !== null
+    );
+  const hasNegativeRegexps = Array.from(searchRegExps.fieldMap.entries()).some(
+    ([_, val]) => val.negative !== null
+  );
```

Comment:
For the record I did this experiment:
```js
let hasPositiveRegexps = Boolean(searchRegexps.generic);
let hasNegativeRegexps = false;

for (const val of searchRegegExps.fieldMap.values()) {
  hasPositiveRegexps ||= val.positive !== null;
  hasNegativeRegexps ||= val.negative !== null;
}
```

Not really longer and IMO simpler to understand ;-)
Indeed I didn't do an early return, as you said they are small maps...
note that Array.from does a full iteration too and you're doing it twice...

It's optional because as you said it's not a bit deal.

Reply:
Yeah, I was thinking about returning early and making etc. It was becoming bigger with that but this seems simple enough. Updated it. (this flow version doesn't understand `||=` so I couldn't use that.)

Reply:
> this flow version doesn't understand ||= so I couldn't use that.)

😑